changelog/change_deprecate_enforce_superclass_module.md

```diff
@@ -0,0 +1 @@
+* [#9087](https://github.com/rubocop-hq/rubocop/pull/9087): Deprecate `EnforceSuperclass` module. ([@koic][])
```

Comment:
Any reason for not using the main changelog?

Reply:
RuboCop core introduces new option to prevent CHANGELOG conflicts. Please see to the bottom of PR checklist for details :-)

> Added an entry (file) to the changelog folder named {change_type}_{change_description}.md if the new code introduces user-observable changes. See changelog entry format for details.

Reply:
Ohh I missed that. Thanks!